src/x/sync/pooled_worker_pool.go

```diff
@@ -94,8 +102,19 @@ func (p *pooledWorkerPool) Go(work Work) {
 	}
 
 	if !p.growOnDemand {
-		workCh <- work
-		return
+		if !onlyIfAvailable {
+			workCh <- work
+			return true
+		}
+
+		// If want to enqueue only if available then determine
+		// whether this can be enqueued or not.
+		select {
```

Comment:
Do you think the semantics might be more obvious if you immediately switched on `onlyIfAvailable`? I'm thinking:

```golang
if onlyIfAvailable {
    select {
    case workCh<- work:
        return true
    default:
    }
    return false
}

... (everything else the same)
```

This way if someone is using the `GoIfAvailable` API it only runs if there appears to be capacity regardless of whether or not `GrowOnDemand` is configured.  This seems less confusing than having the `GoIfAvailable` API basically being a no-op if `GrowOnDemand` is configured.